airbyte-oauth/src/test/java/io/airbyte/oauth/flows/SurveymonkeyOAuthFlowTest.java

```diff
@@ -0,0 +1,115 @@
+    surveymonkeyOAuthFlow = new SurveymonkeyOAuthFlow(configRepository, httpClient, SurveymonkeyOAuthFlowTest::getConstantState);
+
+    workspaceId = UUID.randomUUID();
+    definitionId = UUID.randomUUID();
+    when(configRepository.listSourceOAuthParam()).thenReturn(List.of(new SourceOAuthParameter()
+        .withOauthParameterId(UUID.randomUUID())
+        .withSourceDefinitionId(definitionId)
+        .withWorkspaceId(workspaceId)
+        .withConfiguration(Jsons.jsonNode(ImmutableMap.builder()
+            .put("client_id", "test_client_id")
+            .put("client_secret", "test_client_secret")
+            .build()))));
+    when(configRepository.listDestinationOAuthParam()).thenReturn(List.of(new DestinationOAuthParameter()
+        .withOauthParameterId(UUID.randomUUID())
+        .withDestinationDefinitionId(definitionId)
+        .withWorkspaceId(workspaceId)
+        .withConfiguration(Jsons.jsonNode(ImmutableMap.builder()
+            .put("client_id", "test_client_id")
+            .put("client_secret", "test_client_secret")
+            .build()))));
+  }
+
+  private static String getConstantState() {
+    return "state";
+  }
+
+  @Test
+  public void testGetSourceConcentUrl() throws IOException, InterruptedException, ConfigNotFoundException {
+    final String concentUrl =
+        surveymonkeyOAuthFlow.getSourceConsentUrl(workspaceId, definitionId, REDIRECT_URL);
+    assertEquals(
+        "https://api.surveymonkey.com/oauth/authorize?client_id=test_client_id&redirect_uri=https%3A%2F%2Fairbyte.io&response_type=code&state=state",
+        concentUrl);
+  }
+
+  @Test
+  public void testGetDestinationConcentUrl() throws IOException, InterruptedException, ConfigNotFoundException {
+    final String concentUrl =
+        surveymonkeyOAuthFlow.getDestinationConsentUrl(workspaceId, definitionId, REDIRECT_URL);
+    assertEquals(
+        "https://api.surveymonkey.com/oauth/authorize?client_id=test_client_id&redirect_uri=https%3A%2F%2Fairbyte.io&response_type=code&state=state",
+        concentUrl);
+  }
+
+  @Test
+  public void testCompleteSourceOAuth() throws IOException, InterruptedException, ConfigNotFoundException {
+
+    final Map<String, String> returnedCredentials = Map.of("access_token", "access_token_response");
+    final HttpResponse response = mock(HttpResponse.class);
+    when(response.body()).thenReturn(Jsons.serialize(returnedCredentials));
+    when(httpClient.send(any(), any())).thenReturn(response);
+    final Map<String, Object> queryParams = Map.of("code", "test_code");
+    final Map<String, Object> actualQueryParams =
+        surveymonkeyOAuthFlow.completeSourceOAuth(workspaceId, definitionId, queryParams, REDIRECT_URL);
+
+    assertEquals(Jsons.serialize(Map.of("credentials", returnedCredentials)), Jsons.serialize(actualQueryParams));
+  }
+
+  @Test
+  public void testCompleteDestinationOAuth() throws IOException, ConfigNotFoundException, InterruptedException {
```

Comment:
why are we testing destination? sourcemonkey is only a source ?

Reply:
Trying to test all methods from OAuthFlowImplementation interface.